framer/Animators/Curves.coffee

```diff
@@ -0,0 +1,132 @@
+{BezierCurveAnimator} = require "./BezierCurveAnimator"
+{computeDerivedCurveOptions, computeDuration, computeDampingRatio} = require "./SpringCurveValueConverter"
+{SpringRK4Animator} = require "./SpringRK4Animator"
+{Defaults} = require "../Defaults"
+
+Bezier = (values...) ->
+	(options = {}) ->
+		options.values = values
+		new BezierCurveAnimator(options)
+
+BezierDefaults =
+	linear: Bezier(0, 0, 1, 1)
+	ease: Bezier(.25, .1, .25, 1)
+	easeIn: Bezier(.42, 0, 1, 1)
+	easeOut: Bezier(0, 0, .58, 1)
+	easeInOut: Bezier(.42, 0, .58, 1)
+
+
+Spring = (dampingRatio, mass, velocity) ->
+	curveOptions = {}
+	if dampingRatio? and _.isFinite(dampingRatio)
+		curveOptions.dampingRatio = dampingRatio
+	if mass?
+		curveOptions.mass = mass
+	if velocity?
+		curveOptions.velocity = velocity
+	if not _.isFinite(dampingRatio) and typeof dampingRatio is 'object'
+		curveOptions = dampingRatio
+		if curveOptions.damping? and not curveOptions.dampingRatio?
+			curveOptions.dampingRatio = curveOptions.damping
+	if not curveOptions.tension? and not curveOptions.friction?
+		curveOptions = Defaults.getDefaults "Spring", curveOptions
+
+
+	return (options) ->
+		if curveOptions.dampingRatio?
+			duration = options?.time ? 1
+			derivedOptions = computeDerivedCurveOptions(curveOptions.dampingRatio, duration, curveOptions.velocity, curveOptions.mass)
+			curveOptions = _.defaults derivedOptions, curveOptions
+		else
+			delete options?.time
+		options = _.defaults curveOptions, options
+		animator = new SpringRK4Animator(options)
+		if duration?
+			animator.time = duration
+		animator
+
+_.assign Bezier, BezierDefaults
+Spring.computeDerivedCurveOptions = computeDerivedCurveOptions
+Spring.computeDuration = computeDuration
+Spring.computeDampingRatio = computeDampingRatio
+
+
+exports.Spring = Spring
+exports.Bezier = Bezier
+parseFunction = (string) ->
+	return null unless _.isString string
+
+	regex = /.*(Spring|Bezier)(?:\(\s*{?([a-zA-Z\d:\s,.]*)}?\s*\)|\.(\w+))?/
```

Comment:
Why `a-zA-Z` instead of `\w`?

This regex can use a comment with a bit of explanation of what it tries to achieve.